### PR TITLE
docs(changelog): release v0.16.0 release notes

### PR DESCRIPTION
**Pull request by jm-clius**
Adds the changelog for [Release v0.16.0](https://github.com/status-im/nwaku/milestone/8)

This will be a small maintenance release and I will reduce the milestone scope accordingly.

For the items you were responsible for, please check:
- that it's included in the release notes
- that you're happy with the phrasing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,53 @@
...
+- Refactored and cleaned up peer manager [1539](https://github.com/waku-org/nwaku/pull/1539)
+- Removed unused and legacy websocket submodule [1580](https://github.com/waku-org/nwaku/pull/1580) [1582](https://github.com/waku-org/nwaku/pull/1582)
+- Use base64 URL-safe encoding for noise [1569](https://github.com/waku-org/nwaku/pull/1569)
+- Various general improvements to RLN implementation [1585](https://github.com/waku-org/nwaku/pull/1585) [1587](https://github.com/waku-org/nwaku/pull/1587)
+- Started on implementation for new and improved filter protocol [1584](https://github.com/waku-org/nwaku/pull/1584)
+- Updated pubsub and content topic namespacing to reflect latest changes in RFC [23/WAKU2-TOPICS](https://rfc.vac.dev/spec/23/) [1589](https://github.com/waku-org/nwaku/pull/1589)
+- Unified internal peer data models [1597](https://github.com/waku-org/nwaku/pull/1597)
+- Improved internal implementation of Waku ENR encoding and decoding [1598](https://github.com/waku-org/nwaku/pull/1598) [1599](https://github.com/waku-org/nwaku/pull/1599)
+- Underlying dependency for RLN implementation now loaded as a static library [1578](https://github.com/waku-org/nwaku/pull/1578)
```
Comment (rymnc) [inline CHANGELOG.md:31]: Can we also add updated message validation according to new circuits? ref: https://github.com/waku-org/nwaku/pull/1594

**Review by rymnc — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (jm-clius) [reply CHANGELOG.md:31]: I've added it as a fix for "RLN relay erroneously storing messages from multiple apps". Assuming the PR is wider than that, would you prefer me rather replacing with your suggestion as a `Change`?

Comment (rymnc) [reply CHANGELOG.md:31]: ah, that works too. thanks!

**Pull request merged**